### PR TITLE
edgedb: 3.4.0 -> 4.0.2

### PR DESCRIPTION
**Pull request by ahirner**
* update default.nix src
* update Cargo.lock

## Description of changes

I wonder if `Cargo.lock` can be de-vendored, since it built with `${src}/Cargo.lock` in a flake of mine as well. 
The discussions in https://github.com/NixOS/nixpkgs/pull/221716 go into some depths, but I couldn't find a quick answer.

```
nix shell .#edgedb
edgedb -V
edgedb project init
# ...
edgedb
select "foo";
# {'foo'}
\quit
edgedb instance destroy -I nixpkgs --force
rm -rf dbschema
rm edgedb.toml
```

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/tools/networking/edgedb/default.nix [hunk 1]
```diff
@@ -19,21 +19,21 @@
...
-  version = "3.4.0";
+  version = "4.0.2";
 
   src = fetchFromGitHub {
     owner = "edgedb";
     repo = "edgedb-cli";
-    rev =  "v${version}";
-    sha256 = "sha256-w6YpjSmh517yat45l4gGdV6qWD4O3aCx/6LL5wea+RA=";
+    rev = "v${version}";
+    sha256 = "sha256-uilotat61U6jW1NLh7fVHOujkzUSFRdpeOx+ECGsByY=";
```
Comment (kirillrdy) [inline pkgs/tools/networking/edgedb/default.nix:28]:
```suggestion
    hash = "sha256-uilotat61U6jW1NLh7fVHOujkzUSFRdpeOx+ECGsByY=";
```

File: pkgs/tools/networking/edgedb/default.nix [hunk 2]
```diff
@@ -19,21 +19,21 @@
...
     fetchSubmodules = true;
   };
 
   cargoLock = {
     lockFile = ./Cargo.lock;
```
Comment (kirillrdy) [inline pkgs/tools/networking/edgedb/default.nix:33]: I've asked a question in https://github.com/NixOS/nixpkgs/pull/221716, using Cargo.lock from src seems to be working

**Comment by ahirner** [comment]
Unlike what ofborg suggests, `${src}/Cargo.lock` comes into existence on my machine. I'm not a "nix-pert" enough to say whether it was realized from other builds or so.
```
$ nix-store --query --referrers-closure /nix/store/f1qwbypqxv2xvkqpqf25p4h49j6caaj3-source
/nix/store/f1qwbypqxv2xvkqpqf25p4h49j6caaj3-source
/nix/store/ajfzjh6wlq4dl2l58mzc6dbn7xk412va-cargo-vendor-dir
```

I guess I revert [00dcab4](https://github.com/NixOS/nixpkgs/pull/277193/commits/00dcab45bcdc0c0130ad3806b53d78e883d4b4c9) and then it's good to go? @kirillrdy @ranfdev

**Comment by kirillrdy** [comment]
> Unlike what ofborg suggests, `${src}/Cargo.lock` comes into existence on my machine. I'm not a "nix-pert" enough to say whether it was realized from other builds or so.
> 
> ```
> $ nix-store --query --referrers-closure /nix/store/f1qwbypqxv2xvkqpqf25p4h49j6caaj3-source
> /nix/store/f1qwbypqxv2xvkqpqf25p4h49j6caaj3-source
> /nix/store/ajfzjh6wlq4dl2l58mzc6dbn7xk412va-cargo-vendor-dir
> ```
> 
> I guess I revert [00dcab4](https://github.com/NixOS/nixpkgs/pull/277193/commits/00dcab45bcdc0c0130ad3806b53d78e883d4b4c9) and then it's good to go? @kirillrdy @ranfdev

yeah sorry I forgot to follow up,

"${src}/Cargo.lock" will trigger import from derivation which is disabled in hydra/ofborg to speed up eval time.

you can test this locally by adding
``` 
          allow-import-from-derivation = false
```
into your nix.conf

so yes, revert devendering of lock file

**Comment by ranfdev** [comment]
I'm sorry, I can't follow this. Could you remove me from the maintainers list? Thanks

**Comment by ahirner** [comment]
> maintainers list

Are the thumbs up a vote for emtpy maintainer list or for @kirillrdy to step up? Not sure if an empty list is acceptable ;)

**Comment by kirillrdy** [comment]
> > maintainers list
> 
> Are the thumbs up a vote for emtpy maintainer list or for @kirillrdy to step up? Not sure if an empty list is acceptable ;)

empty list for existing package is acceptable. @ahirner if you are actively using edgedb then I would recommend you to add yourself as maintainer.  I am happy to be added as well. ( if you add yourself or not )

File: pkgs/tools/networking/edgedb/default.nix [hunk 3]
```diff
@@ -68,6 +68,6 @@ rustPlatform.buildRustPackage rec {
     description = "EdgeDB cli";
     homepage = "https://www.edgedb.com/docs/cli/index";
     license = with licenses; [ asl20 /* or */ mit ];
-    maintainers = [ maintainers.ranfdev ];
+    maintainers = with maintainers; [ahirner kirillrd];
```
Comment (kirillrdy) [inline pkgs/tools/networking/edgedb/default.nix:71]:
```suggestion
    maintainers = with maintainers; [ ahirner kirillrdy ];
```

**Comment by ahirner** [comment]
> using edgedb ... happy to be added as well

Sounds great! I know the package but not enough of nix(pkgs).

**Comment by kirillrdy** [comment]
@ahirner can you either squash commits, eg all maintainer related commits squashed.

as for hash change can be own commit ( or squashed ) just need to update commit message so that it matches contribution guidelines

**Comment by kirillrdy** [comment]
apologies, i kinda told you a wrong thing

- maintainers: add ahirner
- edgedb: 3.4.0 -> 4.0.2
- edgedb: update maintainers

File: pkgs/tools/networking/edgedb/default.nix [hunk 4]
```diff
@@ -68,6 +68,6 @@ rustPlatform.buildRustPackage rec {
     description = "EdgeDB cli";
     homepage = "https://www.edgedb.com/docs/cli/index";
     license = with licenses; [ asl20 /* or */ mit ];
-    maintainers = [ maintainers.ranfdev ];
+    maintainers = with maintainers; [ ahirner kirillrd ];
```
Comment (kirillrdy) [inline pkgs/tools/networking/edgedb/default.nix:71]:
```suggestion
    maintainers = with maintainers; [ ahirner kirillrdy ];
```

**Review by kirillrdy — APPROVED** [review]

**Review by thiagokokada — APPROVED** [review]

**Pull request merged**